### PR TITLE
Staging next 2021-12-25

### LINKED ISSUE #152795: Upgrading from 21.11 to unstable fails due to systemd being newer in 21.11
### Describe the bug
Upgrading fails, seemingly because the existing systemd version is newer in 21.11 than unstable.
```
[bweeks@nixos:~]$ sudo nixos-rebuild switch
building Nix...
building the system configuration...
updating systemd-boot from (249.7) to (249.5)
Skipping "/boot/EFI/systemd/systemd-bootx64.efi", since a newer boot loader version exists already.
Skipping "/boot/EFI/BOOT/BOOTX64.EFI", since a newer boot loader version exists already.
Traceback (most recent call last):
  File "/nix/store/4d48b4f5vhskn9qrnd13mpcygpcjh0h4-systemd-boot", line 314, in <module>
    main()
  File "/nix/store/4d48b4f5vhskn9qrnd13mpcygpcjh0h4-systemd-boot", line 267, in main
    subprocess.check_call(["/nix/store/n7shnc1xmriya9djkcvzibw9f6paahxy-systemd-249.5/bin/bootctl", "--path=/boot", "update"])
  File "/nix/store/rppr9s436950i1dlzknbmz40m2xqqnxc-python3-3.9.9/lib/python3.9/subprocess.py", line 373, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/nix/store/n7shnc1xmriya9djkcvzibw9f6paahxy-systemd-249.5/bin/bootctl', '--path=/boot', 'update']' returned non-zero exit status 1.
warning: error(s) occurred while switching to the new configuration

[bweeks@nixos:~]$ 
```

### Steps To Reproduce
1. Install NixOS 21.11
2. Remove 21.11 channel, add unstable channel
3. Run `nixos-rebuild switch`

### Expected behavior
NixOS upgrades from 21.11 to unstable.

### Additional context
84a769c07 is the commit upgrading systemd to 249.7, 50d02797 cherry picks it to 21.11. 

Adding the `--graceful` flag to the `bootctl update` invocation would also solve this issue, albeit leaving the newer EFI files in place.
```console
[nix-shell:~]$ sudo bootctl --path=/boot --graceful update
Skipping "/boot/EFI/systemd/systemd-bootx64.efi", since a newer boot loader version exists already.
Skipping "/boot/EFI/BOOT/BOOTX64.EFI", since a newer boot loader version exists already.
```

### Notify maintainers

@andir @edolstra @flokli @Kloenk 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[bweeks@nixos:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.10-hardened1, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.16`
 - channels(root): `"nixos-22.05pre341453.ac169ec6371"`
 - channels(bweeks): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by veprbl**
constitutents: https://hydra.nixos.org/job/nixpkgs/staging-next/unstable#tabs-constituents
jobset: https://hydra.nixos.org/jobset/nixpkgs/staging-next

Previous staging-next:  #148396

### CODE REVIEW

**Comment by Mindavi** [comment]
https://github.com/NixOS/nixpkgs/pull/150236 should be reverted, it breaks e.g. the guile build (on my end).

**Comment by veprbl** [comment]
- [ ] https://hydra.nixos.org/job/nixpkgs/staging-next/awscli.x86_64-linux @muflax
- [ ] https://hydra.nixos.org/job/nixpkgs/staging-next/pangomm_2_42.x86_64-linux @lovek323 @7c6f434c

File: pkgs/development/python-modules/subunit/default.nix [hunk 1]
```diff
@@ -24,7 +24,7 @@ buildPythonPackage {
   checkInputs = [ testscenarios hypothesis fixtures pytest ];
 
   # requires unittest2, which no longer supported in 3.10
-  doCheck = pythonOlder "3.10";
+  doCheck = pythonAtLeast "3.10";
```
Comment (mweinelt) [inline pkgs/development/python-modules/subunit/default.nix:27]: This change looks odd.

Comment (jonringer) [reply pkgs/development/python-modules/subunit/default.nix:27]: I accidently pushed, when I meant to do a build of the package. But yea, I should be doing stabilization while tired....

File: pkgs/development/python-modules/hydra/default.nix [hunk 2]
```diff
@@ -45,6 +45,15 @@ buildPythonPackage rec {
     "test_config_search_path"
   ];
 
+  disabledTestPaths = [
+    "tests/test_hydra.py"
+  ];
+
+  pythonImportsCheck = [
+    "hydra"
+>>>>>>> origin/master
```
Comment (mweinelt) [inline pkgs/development/python-modules/hydra/default.nix:54]: Try again. @jonringer

Comment (jonringer) [reply pkgs/development/python-modules/hydra/default.nix]: oops

**Comment by fabianhjr** [comment]
Bump on #153795, calibre doesn't build on this staging-next iteration but does build on master.

**Comment by vcunat** [comment]
`python310` [won't build](https://hydra.nixos.org/build/162685449) on x86_64-darwin; that causes the majority of total increase in build failures.  (Total for all platforms including aarch64-darwin now changed roughly like 4k -> 10k.)

**Comment by veprbl** [comment]
#154120

**Pull request merged**